File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Python/Exercises/09_compito_java_20160615/GestionePolisportiva.py
from Squadra import Squadra
from Basket import Basket
from Pallavolo import Pallavolo
from Giocatore import Giocatore

import sys


class GestionePolisportiva:

    def main():

        squadre = []
        cod_to_squadra = {}
        nome_squadra_to_cod = {}
        giocatori = []

        try:
            f = open("squadre.txt", "r")
            line = f.readline().strip()

            while line != "":
                nome_squadra = line
                line = f.readline().strip()
                tok = line.split()
                codice = int(tok[0])
                sport = tok[1]
                line = f.readline().strip()
                tok = line.split()
                vinte = int(tok[0])
                perse = int(tok[1])
                punteggio_medio_numero_medio_set_vinti = float(tok[2])
                line = f.readline().strip()

                nome_squadra_to_cod[nome_squadra] = codice

                if sport == "basket":
                    bas = Basket(nome_squadra, codice, sport, vinte,
                                 perse, punteggio_medio_numero_medio_set_vinti)
                    squadre.append(bas)
                    cod_to_squadra[codice] = bas
                else:
                    pall = Pallavolo(nome_squadra, codice, sport, vinte,
                                     perse, punteggio_medio_numero_medio_set_vinti)
                    squadre.append(pall)
                    cod_to_squadra[codice] = pall

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        print("nome della squadra, codice, n. partite vinte, n. partite perse, punteggio medio, n. medio di set vinti, sport")

        for s in squadre:
            print(s)
        print()

        try:
            f = open("giocatori.txt", "r")
            line = f.readline().strip()

            while line != "":
                tok = line.split()
                codice_squadra = int(tok[0])
                cognome = tok[1]
                eta = int(tok[2])
                maglia = int(tok[3])
                ruolo = tok[4]
                titolare = bool(tok[5])
                gio = Giocatore(codice_squadra, cognome, eta,
                                maglia, ruolo, titolare)
                cod_to_squadra[codice_squadra].add_giocatore(gio)
                giocatori.append(gio)
                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        for sq in squadre:
            sq.stampa_squadra()
        print()

        try:
            nome1 = sys.argv[1]
            nome2 = sys.argv[2]
            nome_squadra_riga_di_comando = nome1 + " " + nome2
            codice_squadra_cercata = nome_squadra_to_cod[nome_squadra_riga_di_comando]

            print("cognome, età, numero di maglia, ruolo, titolare")

            for giocat in giocatori:
                if codice_squadra_cercata == giocat.get_codice():
                    print(giocat)

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)


if __name__ == "__main__":
    GestionePolisportiva.main()

File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/30_compito_java_2019_06_17/GestionePolisportiva.java
import java.io.*;
import java.util.*;

public class GestionePolisportiva {
	static List<Squadra> squadre = new LinkedList<Squadra>();
	static List<Giocatore> giocatori = new LinkedList<Giocatore>();

	public static void main(String[] args) {

		// punto 1

		try {
			BufferedReader br = new BufferedReader(new FileReader("squadre.txt"));
			String line = br.readLine();

			while (line != null) {
				StringTokenizer tok = new StringTokenizer(line);
				int codice = Integer.parseInt(tok.nextToken());
				String sport = tok.nextToken();
				String nomeSquadra = br.readLine();
				line = br.readLine();
				tok = new StringTokenizer(line);
				int vinte = Integer.parseInt(tok.nextToken());
				int perse = Integer.parseInt(tok.nextToken());

				if (sport.equals("pallacanestro")) {
					double falli = Double.parseDouble(tok.nextToken());
					int puntiSegnati = Integer.parseInt(tok.nextToken());
					line = br.readLine();
					Squadra s = new Pallacanestro(codice, sport, nomeSquadra, vinte, perse, falli, puntiSegnati);
					squadre.add(s);
				} else {
					double goalFatti = Double.parseDouble(tok.nextToken());
					line = br.readLine();
					Squadra s = new Pallamano(codice, sport, nomeSquadra, vinte, perse, goalFatti);
					squadre.add(s);

				}

				line = br.readLine();
			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		}

		catch (Exception e) {
			System.err.println(e);
		}

		// punto 2 - memorizzazione giocatori

		try {
			BufferedReader br = new BufferedReader(new FileReader("giocatori.txt"));
			String line = br.readLine();

			while (line != null) {
				int codice = Integer.parseInt(line);
				String cognome = br.readLine();
				String nome = br.readLine();
				line = br.readLine();
				StringTokenizer tok = new StringTokenizer(line);
				boolean titolare = Boolean.parseBoolean(tok.nextToken());
				int eta = Integer.parseInt(tok.nextToken());
				int numeroMaglia = Integer.parseInt(tok.nextToken());
				String ruolo = tok.nextToken();

				Giocatore g = new Giocatore(codice, cognome, nome, titolare, eta, numeroMaglia, ruolo);
				giocatori.add(g);

				line = br.readLine();
			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// punto 3 - stampa tutte le squadre

		System.out.println(
				"nome della squadra\tcodice\tn. partite vinte\tn. partite perse\tpunti totali\tn. medio di falli\tn. medio di reti\tsport");

		for (Squadra s : squadre)
			System.out.println(s);

		// punto 4 - stampa giocatori titolari

		System.out.println("nome\tcognome\teta\tnumero di maglia\truolo\tnome squadra");

		for (Giocatore g : giocatori) {
			if (g.titolare) {
				for (Squadra s : squadre)
					if (g.codiceSquadra == s.codice) {
						System.out.print(g);
						System.out.println(s.nomeSquadra);
					}
			}
		}

		// punto 5 - stampa numero giocatori per squadra

		for (Squadra s : squadre) {
			int giocatoriTotali = 0;
			for (Giocatore g : giocatori) {
				if (s.codice == g.codiceSquadra)
					giocatoriTotali++;
			}
			System.out.println(s.nomeSquadra + " " + giocatoriTotali);
		}
	}
}
